test/libsolidity/util/TestFileParser.h

```diff
@@ -0,0 +1,116 @@
+/*
+	This file is part of solidity.
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+#include <libdevcore/CommonData.h>
+#include <libsolidity/ast/Types.h>
+
+#include <iosfwd>
+#include <stdexcept>
+#include <string>
+#include <vector>
+#include <utility>
+
+namespace dev
+{
+namespace solidity
+{
+namespace test
+{
+
+class TestFileParser
+{
+public:
+	struct ByteFormat
+	{
+		enum Type {
+			UnsignedDec,
+			SignedDec
+		};
+		Type type;
+	};
+
+	struct FunctionCallExpectations
+	{
+		std::string raw;
+		bytes rawBytes;
+		ByteFormat format;
+		bool status = true;
+		std::string output;
+		std::string comment;
+	};
+
+	struct FunctionCallArgs
+	{
+		std::string raw;
+		bytes rawBytes;
+		ByteFormat format;
+		std::string comment;
+	};
+
+	struct FunctionCall
+	{
+		std::string signature;
+		FunctionCallArgs arguments;
+		FunctionCallExpectations expectations;
+		u256 value;
+	};
+
+	static std::string bytesToString(bytes const& _bytes, ByteFormat const& _format);
```

Comment:
Please document. Also, the name sounds too generic. It might also be an option to add that to CommonIO.h.